clang/test/BoundsSafety/Driver/driver.c

```diff
@@ -0,0 +1,11 @@
+// RUN: %clang -c %s -### 2>&1 | FileCheck -check-prefix T0 %s
```

Comment:
`clang/test/Driver/clang_f_opts.c` mixes a lot of options. Newer `-fxxx` tests can go to a dedicated `clang/test/Driver/fxxx.c`

Reply:
@nickdesaulniers I was hoping that we could keep -fbounds-safety tests under a separate folder clang/test/BoundsSafety in order to avoid them from getting mixed up with other tests. And make it easier to run bounds safety related tests only. But I can see this is not what other extensions do so I can move them to the existing layout.

Reply:
@nickdesaulniers As @rapidsna said there are multiple reasons:

1. Provides a very convenient way to run **only `-fbounds-safety` tests**.
2. More likely to avoid merge conflicts with our internal code.

Out of those two I think `(1.)` is the bigger problem because it is something I rely on for local development and I think merge conflicts are still unlikely even if we put tests under their proper location under `clang/test` because we mostly created new tests and tried not to touch existing tests. If `lit` had a way of labelling tests and only running tests with a particular label then the location of the tests would be irrelevant and I would be happy for the tests to live in their normal place under `clang/test/` (e.g. `clang/test/Driver`). Unfortunately AFAIK that feature doesn't exist in lit.

To workaround this would this kind of layout be acceptable?

```
test/Driver/BoundsSafety/...
test/Frontend/BoundsSafety/...
test/Sema/BoundsSafety/...
...
```

That way the tests are conforming to Clang's test layout but its still possible to run **only the -fbounds-safety** tests by doing something like:

```
$ bin/llvm-lit -vs tools/clang/test/{Driver,Frontend,Sema}/BoundsSafety
```

Reply:
Test subdirectories are added conservatively. The main use cases are target-specific tests and flang. `Driver/XRay` is somehow an unneeded use that you can ignore.
```
% echo {CodeGen,Sema,Driver}/*(/)
CodeGen/aarch64_neon_sve_bridge_intrinsics CodeGen/aarch64-sme-intrinsics CodeGen/aarch64-sve2-intrinsics CodeGen/aarch64-sve2p1-intrinsics CodeGen/aarch64-sve-intrinsics CodeGen/arc CodeGen/arm-mve-intrinsics CodeGen/assignment-tracking CodeGen/avr CodeGen/CSKY CodeGen/Inputs CodeGen/LoongArch CodeGen/PowerPC CodeGen/RISCV CodeGen/SystemZ CodeGen/VE CodeGen/WebAssembly CodeGen/X86 Sema/aarch64-sme-intrinsics Sema/aarch64-sve2-intrinsics Sema/aarch64-sve2p1-intrinsics Sema/aarch64-sve-intrinsics Sema/Inputs Driver/flang Driver/Inputs Driver/XRay
```

FWIW you can do

* `path/bin/llvm-lit clang/test/Driver/bounds-safety-*`
* `LIT_OPTS=--filter=bounds-safety path/bin/llvm-lit clang/test/Driver`

and even create an alias for `LIT_OPTS=--filter=bounds-safety `

Reply:
> And make it easier to run bounds safety related tests only.

> Provides a very convenient way to run only -fbounds-safety tests.

Then give each test a similar function name prefix, and then you can do `llvm-lit -vv llvm/test/Driver/fbounds-safety-*`, as @MaskRay said.  Don't start creating new directory structures that don't exist for every other feature implemented by the compiler.

Reply:
@nickdesaulniers @MaskRay Thank you! I removed the new directory and moved the tests to conform to the existing layout.